src/test/e2e/51_oci_compose_test.go

```diff
@@ -109,6 +109,33 @@ func (suite *SkeletonSuite) Test_1_Compose() {
 	suite.NoError(err)
 }
 
+func (suite *SkeletonSuite) Test_2_Component_Templates() {
+	suite.T().Log("E2E: Component Templates")
+	e2e.SetupWithCluster(suite.T())
+	importEverythingPath := fmt.Sprintf("build/zarf-package-import-everything-%s-0.0.1.tar.zst", e2e.Arch)
+
+	_, stdErr, err := e2e.Zarf("package", "inspect", importEverythingPath)
+	suite.NoError(err)
+
+	targets := []string{
+		"import-component-local case was here",
+		"import-component-local-relative case was here",
+		"import-component-wordpress case was here",
+		"import-component-oci case was here",
+		"file-imports case was here",
+		"import-helm-local case was here",
+		"import-helm-local-relative",
+		"import-helm-oci case was here",
+		"import-repos case was here",
+		"import-images case was here",
+	}
+
+	for _, target := range targets {
+		suite.Contains(stdErr, target)
+	}
```

Comment:
this does properly test that component names were replaced, but does not test that the replacement mapped properly

Reply:
this is already on a [per component basis](https://github.com/defenseunicorns/zarf/blob/304d5d7041eaaad7c30cbccd4d9a556f8168592e/src/pkg/packager/variables.go#L159), which uses a [map of strings](https://github.com/defenseunicorns/zarf/blob/304d5d7041eaaad7c30cbccd4d9a556f8168592e/src/pkg/packager/variables.go#L160) to ensure one Component name is mapped each type. 

Do you need more granularity than that said in the above comment?

Reply:
the other hesitation i have around this is the way that we are testing. We are using stdErr which has colors and it not easily unmarshaled to a Component. Would need to change the way we are testing this